datastructure/graphlist.py

```diff
@@ -0,0 +1,45 @@
+class Vertex:
+    def __init__(self, name):
+        self.name = name
+        self.neighbors = set()
+
+    def add_neighbor(self, vertex):
+        self.neighbors.add(vertex)
+
+
+class Graph:
+    vertices = {}
+
+    def add_vertex(self, vertex):
+        if isinstance(vertex, Vertex) and vertex not in self.vertices:
+            self.vertices[vertex.name] = vertex
+            return True
+        else:
+            return False
+
+    def add_edge(self, vertex1, vertex2):
+        if vertex1 in self.vertices and vertex2 in self.vertices:
+            self.vertices[vertex1].add_neighbor(vertex2)
+            self.vertices[vertex2].add_neighbor(vertex1)
+            return True
+        else:
+            return False
+
```

Comment:
Remove edge, remove vertex (which should remove anything  any edges that point to it)?

Reply:
Okay, I will write some methods for those. 
Is there any other methods I should add?

Reply:
graphs are simple in its components, whats hard are its algorithms and problems. I dont know what other build it methods to put in right now.